crates/runtime/tests/graphql/mod.rs

```diff
@@ -209,12 +209,26 @@ async fn start_server() -> Result<(tokio::sync::oneshot::Sender<()>, SocketAddr)
     Ok((tx, addr))
 }
 
-fn make_graphql_dataset(path: &str, name: &str, query: &str, json_pointer: &str) -> Dataset {
+fn make_graphql_dataset(
+    path: &str,
+    name: &str,
+    query: &str,
+    json_pointer: &str,
+    unnest_depth: Option<u32>,
+) -> Dataset {
     let mut dataset = Dataset::new(format!("graphql:{path}"), name.to_string());
-    let params = HashMap::from([
-        ("json_pointer".to_string(), json_pointer.to_string()),
-        ("graphql_query".to_string(), query.to_string()),
-    ]);
+    let params = if let Some(unnest_depth) = unnest_depth {
+        HashMap::from([
+            ("json_pointer".to_string(), json_pointer.to_string()),
+            ("graphql_query".to_string(), query.to_string()),
+            ("unnest_depth".to_string(), unnest_depth.to_string()),
```

Comment:
Might be better to make params mut and add unnest_depth separately to avid duplicate code.